cmd/worker/main.go

```diff
@@ -36,7 +36,66 @@ const (
 	localPkgPathFmt = "/local/%s"
 )
 
-func handleMessage(ctx context.Context, msg *pubsub.Message, packagesBucket *blob.Bucket, resultsBucket, fileWritesBucket, imageTag string, notificationTopic *pubsub.Topic) error {
+// resultBucketPaths holds bucket paths for the different types of results
+type resultBucketPaths struct {
+	dynamicAnalysis string
+	staticAnalysis  string
+	fileWrites      string
+}
+
+func copyPackageToLocalFile(ctx context.Context, packagesBucket *blob.Bucket, bucketPath string) (string, *os.File, error) {
+	if packagesBucket == nil {
+		return "", nil, errors.New("packages bucket not set")
+	}
+
+	// Copy remote package path to temporary file.
+	r, err := packagesBucket.NewReader(ctx, bucketPath, nil)
+	if err != nil {
+		return "", nil, err
+	}
+	defer r.Close()
+
+	f, err := os.CreateTemp("", "")
+	if err != nil {
+		return "", nil, err
+	}
+	defer os.Remove(f.Name())
```

Comment:
Is this correct? This would be deleting the file that we need as soon as this function returns right? 

I think the `defer` would need to move out of this function.

Reply:
Good catch, fixed